### ISSUE TITLE
float[] is very slow

### ISSUE THREAD

**Issue by kirillp** [gwtproject/gwt#9501]
**GWT version:** 2.8 
**Browser (with version):** any
**Operating System:** any

The problem is that JS Array significantly slower then typed array.
Also it produces lot of garbage on some browsers due to boxing. 
Calculations on large float[], int[], double[] show significant performance and memory improvements when using typed arrays. 
Gwt should consider adding a compiler option to use platform arrays for arrays of primitive types instead of JS arrays.

It is very important because today performance critical WebGL applications become more popular and growing (+WebGL2, WebVR). 

##### Known workarounds

As a known workaround one can use the following code:

  public static native float[] createFastFloat(int size) /*-{
      return new Float32Array(size);
  }-*/;

However this approach is not portable.

**Comment by mincong-h** [comment]
Package [`com.google.gwt.typedarrays`][typedarrays] provides support for different typed arrays. For example, interface [`Float32Array.java`][Float32Array] has 2 implementations:

- [`com.google.gwt.typedarrays.server.Float32ArrayImpl`][Float32ArrayImpl] for the server side
- [`com.google.gwt.typedarrays.client.Float32ArrayNative`][Float32ArrayNative] for the client side

[typedarrays]: https://github.com/gwtproject/gwt/tree/2.8.2/user/src/com/google/gwt/typedarrays
[Float32Array]: https://github.com/gwtproject/gwt/blob/2.8.2/user/src/com/google/gwt/typedarrays/shared/Float32Array.java
[Float32ArrayImpl]: https://github.com/gwtproject/gwt/blob/2.8.2/user/src/com/google/gwt/typedarrays/server/Float32ArrayImpl.java
[Float32ArrayNative]: https://github.com/gwtproject/gwt/blob/2.8.2/user/src/com/google/gwt/typedarrays/client/Float32ArrayNative.java

Do they fit your need?

**Comment by kirillp** [comment]
No, they are not portable. Here is a different issue. Assume we already have a portable (jvm, teavm, android) code that compute something on an array. The problem is that the GWT compilation result will be very slow.

Also linking gwt-server to an Android app is not a good idea. And the performance of Float32ArrayImpl is very slow on mobile comparing to a plain array.
